### PR TITLE
Feat/document public classes

### PR DESCRIPTION
**Pull request by elkesrio**

### CODE REVIEW

**Review by Pierre-Michard — APPROVED** [review]

**Review by glavigno — APPROVED** [review]

File: lib/src/common/logger/common_logger.dart [hunk 1]
```diff
@@ -35,7 +35,7 @@ void _printOutput({
 }
 
 ///
-/// Enable the logs to be display in the console.
+/// Enables the logs to be displayed in the console.
```
[the remark this answers is not in the document]
Comment (elkesrio) [reply lib/src/common/logger/common_logger.dart:38]: ah oui non tu as raison. Bien vu

File: lib/src/keystore/impl/keystore.dart [hunk 2]
```diff
@@ -5,30 +5,43 @@ import 'package:equatable/equatable.dart';
...
 // we hide Prefixes from the crypto name and then show it without a name to avoid this:
-// `static const crypto.Prefixes seedPrefix = crypto.Prefixes.edsk2;`
+// `static const crypto.Prefixes _seedPrefix = crypto.Prefixes.edsk2;`
 // and have this instead :
 // `static const Prefixes = Prefixes.edsk2;
 import 'package:tezart/src/crypto/crypto.dart' as crypto hide Prefixes;
 import 'package:tezart/src/crypto/crypto.dart' show Prefixes;
 import 'package:tezart/src/signature/signature.dart';
 
+/// A class that handles key stores and keys (seed, secret key, public key, address) computation.
```
Comment (akettal) [inline lib/src/keystore/impl/keystore.dart:15]:
```suggestion
/// A class that handles keystores and keys (seed, secret key, public key, address) computation.
```

File: lib/src/keystore/impl/keystore.dart [hunk 3]
```diff
@@ -5,30 +5,43 @@ import 'package:equatable/equatable.dart';
...
+  static const _secretKeyLength = 98;
+  static const _seedLength = 54;
 
   final String secretKey;
 
   final String mnemonic;
 
   const Keystore._({@required this.secretKey, this.mnemonic});
 
+  /// A factory that generates a key store from a secret key.
```
Comment (akettal) [inline lib/src/keystore/impl/keystore.dart:33]:
```suggestion
  /// A factory that generates a keystore from a secret key.
```

File: lib/src/keystore/impl/keystore.dart [hunk 4]
```diff
@@ -37,9 +50,18 @@ class Keystore extends Equatable {
     });
   }
 
+  /// A factory that generates a key store from a seed.
```
Comment (akettal) [inline lib/src/keystore/impl/keystore.dart:53]:
```suggestion
  /// A factory that generates a keystore from a seed.
```

File: lib/src/keystore/impl/keystore.dart [hunk 5]
```diff
@@ -48,11 +70,19 @@ class Keystore extends Equatable {
     });
   }
 
+  /// A factory that generates a key store from a mnemonic, email and password.\
```
Comment (akettal) [inline lib/src/keystore/impl/keystore.dart:73]:
```suggestion
  /// A factory that generates a key store from a mnemonic, email and password (for fundraisers).\
```

**Pull request merged**